File: two_sample.c
/* Nathan Englehart (Summer, 2022) */

#include <stdio.h>
#include <math.h>

double two_sample(double x_bar_1, double x_bar_2, double s_1, double s_2, int n_1, int n_2) {

	/* Returns t value for two tailed t-test using given variables. */

	return (x_bar_1 - x_bar_2) / (sqrt((pow(s_1,2)/n_1-1) + ((pow(s_2,2)/n_2-1))));

}

int main() {

	/* Variables */

	double x_bar_1 = 3392.00;
	double x_bar_2 = 16610.86;
	double s_1 = 3848.102;
	double s_2 = 3725.971;
	int n_1 = 84;
	int n_2 = 21;

	double t = fabs(two_sample(x_bar_1, x_bar_2, s_1, s_2, n_1, n_2));

	printf("absolute t = %f.\n",t);
	printf("df = %d.\n",n_1 + n_2 - 2);

	return 0;

}

File: two_sample.r
#!/usr/bin/env Rscript

### Nathan Englehart (Summer 2022)

library(haven) # to read sav

gss_data <- read_sav("data.sav")

two_sample_t_test <- function(sample_1,sample_2) {

   ### Returns the t value and degrees of freedom with two sample t test using given samples. 
   ###	
   ###   Args:
   ###          
   ###	      sample_1::[List]
   ###	         First given sample 
   ###
   ###	      sample_2::[List]
   ###	         Second given sample
   ###
   ###

   t = (mean(sample_1, na.rm = TRUE) - mean(sample_2, na.rm = TRUE))/sqrt((((sd(sample_1, na.rm = TRUE))^2)/(sum(!is.na(sample_1))-1)) + (((sd(sample_2, na.rm = TRUE))^2)/(sum(!is.na(sample_2))-1)))
   df = sum(!is.na(sample_1)) + sum(!is.na(sample_2)) - 2

   return(list(t,df))
}

s <- as.numeric(unlist(gss_data[,"sex"]))
t <- as.numeric(unlist(gss_data[,"tvhours"]))

sample <- data.frame(s,t)

m <- subset(sample, sample$s == 1)
w <- subset(sample, sample$s == 2)

m_hours <- m$t
w_hours <- w$t

result <- two_sample_t_test(m_hours,w_hours)
result
